R/translate-sql-odbc.R

```diff
@@ -35,7 +50,79 @@ base_odbc_agg <- sql_translator(
 base_odbc_win <- sql_translator(
   .parent = base_win,
   n             = function() sql("COUNT(*)"),
-  count         = function() sql("COUNT(*)")
+  count         = function() sql("COUNT(*)"),
+
+  # Window functions in this variant use field that its being
+  # 'windowed' as the secondary ORDER BY instead of the field
+  # from win_current_order().  In the tests, having an ORDER BY
```

Comment:
Oh ok, I was attempting to match what we can do with dplyr locally, like: 
```
 mtcars %>%
  mutate(next_mpg = lead(mpg)) 
```
Also, I did try using `arrange()` but it fails for me on both PR and CRAN versions.  Mainly because `arrange()` creates a subquery with a`ORDER BY`, instead of adding the `ORDER BY` to the partition clause  .Here are the errors I get:

Code:
```
m <- tbl(con, "mtcars") %>%
  arrange(mpg) %>%
  mutate(next_mpg = lead(mpg))

m
```

CRAN:

```
<SQL>
SELECT "row_names", "mpg", "cyl", "disp", "hp", "drat", 
"wt", "qsec", "vs", "am", "gear", "carb", LEAD("mpg") AS "next_mpg"
FROM (SELECT *
FROM "mtcars"
ORDER BY "mpg") "jllnbfdoyt"
```
```
 Show Traceback
Error in new_result(connection@ptr, statement) : nanodbc/nanodbc.cpp:1344: HY000:
 The function 'LEAD' must have an OVER clause., Incorrect syntax near the keyword 
'ORDER'., Statement(s) could not be prepared.,
```

PR:

```
Error in new_result(connection@ptr, statement) : nanodbc/nanodbc.cpp:1344: HY000: 
The ORDER BY clause is invalid in views, inline functions, derived tables, subqueries,
 and common table expressions, unless TOP, OFFSET or FOR XML is also specified., 
Statement(s) could not be prepared.,
```